File: app/src/main/java/us/shiroyama/wireless/BluetoothDeviceAdapter.java
package us.shiroyama.wireless;

import android.annotation.SuppressLint;
import android.bluetooth.BluetoothDevice;

import androidx.annotation.NonNull;

import java.util.List;

public class BluetoothDeviceAdapter extends DeviceAdapter<BluetoothDevice> {
    public BluetoothDeviceAdapter(@NonNull List<BluetoothDevice> deviceList) {
        super(deviceList);
    }

    @Override
    public void onBindViewHolder(@NonNull DeviceViewHolder holder, int position) {
        BluetoothDevice device = deviceList.get(position);
        @SuppressLint("MissingPermission")
        String name = device.getName() == null ? "-------" : device.getName();
        @SuppressLint("MissingPermission")
        int type = device.getType();
        String typeString = "";
        switch (type) {
            case BluetoothDevice.DEVICE_TYPE_CLASSIC: {
                typeString = "Bluetooth Classic";
                break;
            }
            case BluetoothDevice.DEVICE_TYPE_DUAL: {
                typeString = "Bluetooth Dual";
                break;
            }
            case BluetoothDevice.DEVICE_TYPE_LE: {
                typeString = "Bluetooth Low Energy";
                break;
            }
            default: {
                typeString = "Bluetooth Unknown";
                break;
            }
        }
        holder.textViewDevice.setText(name);
        holder.textViewAddress.setText(device.getAddress());
        holder.textViewType.setText(typeString);
    }
}

File: app/src/main/java/us/shiroyama/wireless/BluetoothDeviceAdapter.kt
package us.shiroyama.wireless

import android.annotation.SuppressLint
import android.bluetooth.BluetoothDevice

class BluetoothDeviceAdapter(deviceList: List<BluetoothDevice>) :
    DeviceAdapter<BluetoothDevice>(deviceList) {
    override fun onBindViewHolder(holder: DeviceViewHolder, position: Int) {
        val device = deviceList[position]
        @SuppressLint("MissingPermission") val name =
            if (device.name == null) "-------" else device.name
        @SuppressLint("MissingPermission") val type = device.type
        var typeString = ""
        typeString = when (type) {
            BluetoothDevice.DEVICE_TYPE_CLASSIC -> {
                "Bluetooth Classic"
            }
            BluetoothDevice.DEVICE_TYPE_DUAL -> {
                "Bluetooth Dual"
            }
            BluetoothDevice.DEVICE_TYPE_LE -> {
                "Bluetooth Low Energy"
            }
            else -> {
                "Bluetooth Unknown"
            }
        }
        holder.textViewDevice.text = name
        holder.textViewAddress.text = device.address
        holder.textViewType.text = typeString
    }
}